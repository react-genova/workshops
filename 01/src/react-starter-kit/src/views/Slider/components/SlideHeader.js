//@flow
import * as React from 'react';
import styled from 'styled-components';

const Container = styled.div`
    display: flex;
    flex-direction: column;
    font-family: Arial, Helvetica, sans-serif;
    background-color: #264d73;
    align-content: center;
    justify-content: center;
    align-items: center;
    padding: 0.6em;
`;

const Title = styled.div`
    font-size: 2em;
    font-weight: bold;
    color: white;
`;

const SubTitle = styled.div`
    font-size: 1.5em;
    color: #A0A0A0;
`;

type Props = {
    title:string,
    description:string,
};

const SlideHeader = ({title, description}:Props) => (
    <Container>
        <Title>{title}</Title>
        <SubTitle>{description}</SubTitle>
    </Container>
);

export default SlideHeader;