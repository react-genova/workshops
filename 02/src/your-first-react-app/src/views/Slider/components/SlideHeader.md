```js
<SlideHeader title="react-genova" description="workshop 01" />
```