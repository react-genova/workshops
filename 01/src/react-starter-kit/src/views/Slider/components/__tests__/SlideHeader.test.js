//@flow
import React from 'react';
import SlideHeader from '../SlideHeader';
import { shallow } from 'enzyme';

describe('SlideHeader component', () => {

    it('simple header snapshot', () => {
        const component = shallow(<SlideHeader title="react-genova" description="workshop 01" />);
        expect(component).toMatchSnapshot();
    });

});