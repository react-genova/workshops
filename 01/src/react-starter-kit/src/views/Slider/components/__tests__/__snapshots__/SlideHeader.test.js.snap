// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideHeader component simple header snapshot 1`] = `
<styled.div>
  <styled.div>
    react-genova
  </styled.div>
  <styled.div>
    workshop 01
  </styled.div>
</styled.div>
`;
